spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfiguration.java

```diff
@@ -53,22 +53,23 @@
  * @author Vince Bickers
  * @author Stephane Nicoll
  * @author Kazuki Shimizu
+ * @author Michael Simons
  * @since 1.4.0
  */
 @Configuration
 @ConditionalOnClass({ SessionFactory.class, Neo4jTransactionManager.class, PlatformTransactionManager.class })
-@ConditionalOnMissingBean(SessionFactory.class)
 @EnableConfigurationProperties(Neo4jProperties.class)
 @Import(Neo4jBookmarkManagementConfiguration.class)
 public class Neo4jDataAutoConfiguration {
 
 	@Bean
-	@ConditionalOnMissingBean
+	@ConditionalOnMissingBean({org.neo4j.ogm.config.Configuration.class, SessionFactory.class})
 	public org.neo4j.ogm.config.Configuration configuration(Neo4jProperties properties) {
 		return properties.createConfiguration();
 	}
 
 	@Bean
+	@ConditionalOnMissingBean(SessionFactory.class)
```

Comment:
How about moving those two bean definitions to a dedicated class? Can be an inner class here. That way we could move the condition on the `SessionFactory` there and limit the change to a minimum.

The name of that inner class would also convey the use case we're trying to bring here.

Reply:
Nice idea. I did this, but had to move the private helper method `getPackagesToScan`, too.